tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -434,20 +437,57 @@ func (iptConfigurator *IptablesConfigurator) run() {
 	iptConfigurator.handleInboundIpv4Rules(ipv4RangesInclude)
 	iptConfigurator.handleInboundIpv6Rules(ipv6RangesExclude, ipv6RangesInclude)
 
-	// Execute iptables commands
-	for _, cmd := range iptConfigurator.iptables.BuildV4() {
-		if len(cmd) > 1 {
-			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
-		} else {
-			iptConfigurator.ext.RunOrFail(cmd[0])
-		}
+	iptConfigurator.executeCommands()
+}
+
+func (iptConfigurator *IptablesConfigurator) createRulesFile(filePath string, contents string) {
+	f, err := os.Create(filePath)
+	if err != nil {
+		fmt.Println("Unable to create file for iptables-restore")
+		panic(err)
+	}
+	defer f.Close()
+	writer := bufio.NewWriter(f)
+	_, err = writer.WriteString(contents)
+	if err != nil {
+		fmt.Println("Unable to write file for iptables-restore")
+		panic(err)
 	}
+	writer.Flush()
+}
 
-	for _, cmd := range iptConfigurator.iptables.BuildV6() {
+func (iptConfigurator *IptablesConfigurator) executeIptablesCommands(commands [][]string) {
+	for _, cmd := range commands {
 		if len(cmd) > 1 {
 			iptConfigurator.ext.RunOrFail(cmd[0], cmd[1:]...)
 		} else {
 			iptConfigurator.ext.RunOrFail(cmd[0])
 		}
 	}
 }
+
+func (iptConfigurator *IptablesConfigurator) executeCommands() {
+	if iptConfigurator.cfg.RestoreFormat {
```

Comment:
I would find it more readable to group the `iptables-restore` code for both IPv4 and IPv6 together:
```
	if iptConfigurator.cfg.RestoreFormat {
		// IPv4
		// IPv6
	} else {
		// IPv4
		// IPv6
	}
```

Reply:
Nice!